### ISSUE TITLE
How to keep presenter surviving when set "Don't keep activities"?

### ISSUE THREAD

**Issue by luoyushouchai** [czyrux/MvpLoaderSample#14]
When i set "**Don't keep activities**",  The loader will be null when presenter-activity is in background!

**Comment by SufianBabri** [comment]
It happens because your Activity's `onStop()` has been called.

You need to find a way to arrange your flow so it respects the MVP architecture, not the other way around.

To get a more relatable answer, you need to explain a little bit of what you're trying to do.

**Comment by luoyushouchai** [comment]
The Android framework may destroy  activity any time it's in the background or backstack, I want to make my activity show correctly when this happens.  The "don't keep activities" developer option simply changes the framework's behavior so it will always destroy your activity when it goes into the background or backstack. This allows us to test how your activity responds to what is normally a rare occurrence.
    I use the loader to test on config changes, it work well! 
    When i set developer option "Don't keep activities", 
   ```
 Loader<P> loader = getSupportLoaderManager().getLoader(loaderId());
  ```

loader is null , when i go to secondActivity, and back!
I can not restore presenter on this condition.

**Comment by SufianBabri** [comment]
Yes this is expected behaviour. MVP will not help you in making the Activity or its Loader unkillable.

This version of MVP will help you keep your Presenter when you rotate your device under normal conditions. Once the Activity is destroyed (e.g. closed under low memory situation, which can be done by the option "Don't keep activities") its loader will also get destroyed. There's no way around this other than writing to a SharedPreferences, DB or disk (but then you'd have to find a way to accurately determine when was the Activity destroyed but not killed, so that you can restore its previous loader/data).

**Comment by luoyushouchai** [comment]
Thanks ! Understand it!

**Comment by SufianBabri** [comment]
Another important point: don't use Presenters to store data of the screen.
Use `onSaveInstanceState(Bundle outState)` of Activity and Fragment instead. (It's a lifecycle method and the Android will call it itself. You should NEVER call it by yourself!)

**Comment by luoyushouchai** [comment]
MyPresenter has some datasources, some of whichs keeping data in cache, so I want to keep my presenter when low memory situation!  
Is there a good suggestion?

**Comment by SufianBabri** [comment]
If you want things to maintain them beyond lifecycles of Activity/Fragment then you could use DB (SQLite, Realm, etc), SharedPreferences or any using File system.

You're fighting against the OS by trying to keep the Presenter under low memory, and that will only end up with the other guy winning it. 😉

**Comment by luoyushouchai** [comment]
@SufianBabri . Thank you a lot!
